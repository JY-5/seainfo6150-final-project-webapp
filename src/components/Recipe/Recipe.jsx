import React from 'react';
import { useEffect } from 'react';
import { useLocation } from "react-router-dom";
import styles from "./Recipe.module.css";
import roastedChicken1 from "../../assets/images/roastedChicken1.jpg";
import roastedChicken2 from "../../assets/images/roastedChicken2.jpg";
import roastedChicken3 from "../../assets/images/roastedChicken3.jpg";

const Recipe = ({ recipe }) => {
    const { pathname } = useLocation();
    useEffect(() => {
        window.scrollTo(0, 0);
    }, [pathname]);
      
    return (
        <div className={styles.container}>
            {recipe ?
                (<>
                <h2 className={styles.title}>{recipe["title"]}</h2>
                <div>
                    {(recipe.id === "1") ?
                    (<img className={styles.responsive}
                        srcSet={roastedChicken3 + " 300w, " + roastedChicken2 + " 600w," + roastedChicken1 + " 1200w"} 
                        sizes="(max-width: 320px) 70vw,
                                (max-width: 500px) 70vw,
                                (max-width: 600px) 50vw,
                                30vw"
                        src={roastedChicken3} alt={recipe["title"]}/>)
                    : (<div className={styles.imageContainer}>
                        <img className={styles.image} src={recipe.url} alt={recipe["title"]}/>
                        </div>
                        )}
                    <h3>Ingredients</h3>
                    <p>{recipe["ingredients"]}</p>
                </div>
                <div>
                <h3>Instructions</h3>
                <ul>
                {Object.keys(recipe["instructions"]).map(step => 
                    (
                        <li key={step}>
                            <h4>{step}</h4>
                            <p>{recipe["instructions"][step]}</p>
                        </li>
                    )
                )}
                </ul>
            </div>
            </>)
            : null
            }
        </div>
    )
}

export default Recipe

