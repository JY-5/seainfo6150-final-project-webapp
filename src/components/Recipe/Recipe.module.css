.container {
    margin: 2rem;
    margin-top: 14.5rem;
    font-family: 'Rubik', sans-serif;
}

.title {
    margin-top: 6rem;
    font-family: 'Prompt', sans-serif;
    color: #539222;
}

.image {
    width: 20rem;
    height: 20rem;
    border-radius: 5px;
}

.responsive {
    border-radius: 5px;
}

.container li {
    list-style: none;
}

.container ul {
    padding-left: 0;
}

@media screen and (min-width: 768px) {
    .container {
        margin-top: 7rem;
    }
}