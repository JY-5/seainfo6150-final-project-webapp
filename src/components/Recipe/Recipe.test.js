import React from "react";
import Recipe from './Recipe';
import { render } from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";

const recipe =  {
    "title": "Salmon Steak",
    "id": "11",
    "url": "https://www.cookingclassy.com/wp-content/uploads/2017/02/skillet-seared-salmon-2.jpg",
    "ingredients": "300 gm salmon fish, 1 cup onion, 2 tablespoon lemon pepper seasoning, 4 tablespoon butter, 4 tablespoon lemon juice, garlic salt as required, 4 slices lemon",
    "instructions": {
        "Step 1": "To make this delicious recipe, wash the salmon under running water. Then drain the excess water and cut them into desired shapes.",
        "Step 2": "Take a bowl, add 2 tablespoon butter and 2 tablespoon lemon juice with a dash of salt. Once the fish is moist, marinate the fish and keep it aside for sometime. In the meantime, preheat the oven at 180 degree Celsius.",
        "Step 3": "Next, grease the baking tray with some butter. Place the marinated fish and pour the remaining butter and lemon juice mixture with some chopped coriander and spices. Sprinkle garlic salt and lemon pepper seasoning. Place the lemon slices over the pieces and bake for 20 minutes at 200 degree Celsius.",
        "Step 4": "Once the fish is baked, you can serve it with some fresh coriander leaves, tomatoes and spring onions. Your Salmon Steak is ready and indulge in the goodness."
    }
};

describe("Add Recipe tests", () => {
    it("renders correctly", () => {
        jest.mock('react-router-dom', () => ({
            ...jest.requireActual('react-router-dom'),
            useLocation: jest.fn()
        }));
        const { container } = render(<Router><Recipe recipe={recipe}/></Router>);
        expect(container).toMatchSnapshot();
    });
});