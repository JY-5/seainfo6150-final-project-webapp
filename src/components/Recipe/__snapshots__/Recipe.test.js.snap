// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Recipe tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      Salmon Steak
    </h2>
    <div>
      <img
        alt="Salmon Steak"
        class="image"
        src="https://www.cookingclassy.com/wp-content/uploads/2017/02/skillet-seared-salmon-2.jpg"
      />
      <h3>
        Ingredients
      </h3>
      <p>
        300 gm salmon fish, 1 cup onion, 2 tablespoon lemon pepper seasoning, 4 tablespoon butter, 4 tablespoon lemon juice, garlic salt as required, 4 slices lemon
      </p>
    </div>
    <div>
      <h3>
        Instructions
      </h3>
      <ul>
        <li>
          <h4>
            Step 1
          </h4>
          <p>
            To make this delicious recipe, wash the salmon under running water. Then drain the excess water and cut them into desired shapes.
          </p>
        </li>
        <li>
          <h4>
            Step 2
          </h4>
          <p>
            Take a bowl, add 2 tablespoon butter and 2 tablespoon lemon juice with a dash of salt. Once the fish is moist, marinate the fish and keep it aside for sometime. In the meantime, preheat the oven at 180 degree Celsius.
          </p>
        </li>
        <li>
          <h4>
            Step 3
          </h4>
          <p>
            Next, grease the baking tray with some butter. Place the marinated fish and pour the remaining butter and lemon juice mixture with some chopped coriander and spices. Sprinkle garlic salt and lemon pepper seasoning. Place the lemon slices over the pieces and bake for 20 minutes at 200 degree Celsius.
          </p>
        </li>
        <li>
          <h4>
            Step 4
          </h4>
          <p>
            Once the fish is baked, you can serve it with some fresh coriander leaves, tomatoes and spring onions. Your Salmon Steak is ready and indulge in the goodness.
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
